api/src/main/java/org/eclipse/microprofile/metrics/Snapshot.java

```diff
@@ -110,4 +118,50 @@ public String toString() {
         }
 
     }
+
+    /**
+     * Represents a cumulative histogram bucket at the moment it was sampled from the Snapshot. The bucket of
+     * {@link Timer} will be represented in nanoseconds.
+     *
+     * See {@link #bucketValues()}
+     */
+    public static class HistogramBucket {
+        private final double bucket;
+        private final double count;
```

Comment:
Given we don't have any use for count to be non-integer, I suggest we switch to using long here.  Impls can format as a float, if desired, for Prometheus.

Reply:
I thought I had posted this earlier here...maybe I did so elsewhere...but FWIW https://github.com/micrometer-metrics/micrometer/issues/1925 discusses Micrometer's choice of `double` for values like this.

Reply:
That's a really useful reference. 

Micrometer uses doubles for Counters -- which have a `void increment(double amount)` method, and generally treat all counts as doubles.  On the other hand, MP Metrics uses longs for Counters -- which have a `void inc(long n)` method). 

That link mentions a few reasons for using doubles.  "Measuring partial changes" doesn't apply for histogram buckets, since each item added to a Histogram counts as exactly 1 item added to a bucket.  "Maximum precision" may be true, but longs are also very precise. "Maximum compatibility with underlying metrics libraries" is a good point, but we've already made the choice to have Counters be represented by longs (so we're already dealing with the double/long conversions).

Given all that, I still think we should choose "long" for our bucket counts, partly to be consistent with how we count things elsewhere in our API (specifically in Counters), and also because histogram bucket counts are always integer values (so it would be a strange place to choose to be inconsistent).

Reply:
@tjquinno ^^

Reply:
I did see this and am convinced by Don's impeccable reasoning. Sorry, I should have noted as much here.